zipline/finance/blotter/core.py

```diff
@@ -0,0 +1,111 @@
+from functools import partial
+from .blotter import Blotter
+from zipline.utils.compat import mappingproxy
+
+
+class BlotterClassDispatcher(object):
+    """
+    A class for registering and dispatching custom blotter classes.
+
+    Method of a global instance of this class are provided by
+    zipline.utils.calendar_utils.
+
+    Parameters
+    ----------
+    blotter_factories : dict[str -> function]
+        Factories for lazy blotter creation.
+    """
+    def __init__(self, blotter_factories):
+        self._blotter_factories = blotter_factories
+        self.blotter_factories = mappingproxy(self._blotter_factories)
+
+    def load(self, name):
+        """
+        Retrieves a blotter whose name is given.
+
+        Parameters
+        ----------
+        name : str
+            The name of the blotter class to be retrieved.
+
+        Returns
+        -------
+        blotter_class : zipline.finance.blotter.Blotter
+            The desired blotter class.
+        """
+        try:
+            return self._blotter_factories[name]
+        except KeyError:
+            raise ValueError("no blotter class registered as %r, options "
+                             "are: %r" % (name, sorted(self._blotter_factories
+                                                       )))
+
+    def class_exists(self, name):
+        """
+        Whether or not the global list of blotter classes contains the
+        class with the specified name
+
+        :param name: The name of the blotter class
```

Comment:
we use numpy doc style instead of normal sphinx docstrings. This would look like:

```
Parameters
----------
name : str
    The name of the blotter class.

Returns
-------
exists : bool
    Is this name registered to a blotter class?
```